neo/Network/P2P/Payloads/BlockBase.cs

```diff
@@ -59,8 +59,9 @@ Witness[] IVerifiable.Witnesses
         public virtual void Deserialize(BinaryReader reader)
         {
             ((IVerifiable)this).DeserializeUnsigned(reader);
-            if (reader.ReadByte() != 1) throw new FormatException();
-            Witness = reader.ReadSerializable<Witness>();
+            Witness[] witnesses = reader.ReadSerializableArray<Witness>(1);
```

Comment:
Why not ?
```csharp
            Witness witnesses = reader.ReadSerializable<Witness>();
```

Reply:
We can do it.. the problem I see is only compatibility to JSON spec serialization. Will we change it to "witness" too? As long as we do it, it's fine for me. It's like having an inheritance that removed virtualization  from Witnesses and exposed Witness. No logical issue I guess.

Reply:
I agree @shargon, no need to waste this byte for nothing. Lets update json spec too, to make it compatible.

Reply:
No. We need to keep it as an array. So we can allow multiple witnesses in the future.

Reply:
Is not better to increase the version if we want to modify this in the future?

Reply:
> Is not better to increase the version if we want to modify this in the future?

That sounds good. Otherwise, why have a version 🤷‍♂

Reply:
Do you think there's hope for using that in blocks Erik? I thought about it, and perhaps, there's no more applications for multi witness here on Neo3.... please correct me if I'm mistaken.

Reply:
Yes. If we support grouped consensus. And I think that even if we don't need multiple `Witness` in the future, we should serialize it according to the definition of the `IVerifiable.Witnesses` interface.

Reply:
Perfect. I fully agree with both justifications Erik.